nvflare/app_opt/xgboost/histogram_based_v2/executor.py

```diff
@@ -85,10 +85,13 @@ def handle_event(self, event_type: str, fl_ctx: FLContext):
             adaptor.set_abort_signal(self.abort_signal)
             adaptor.initialize(fl_ctx)
             self.adaptor = adaptor
+        elif event_type == Constant.EVENT_XGB_ABORTED:
```

Comment:
Since we changed SecurityHandler's _abort() method to simply fire this event (and not call system_panic()), and SecurityHandler is also inherited by ServerHandler (and it also calls self._abort), we need to update the XGBController to also catch this event and call system_panic(). 

 Make sure you call super().handle_event() from XGBController's event handler since its parent Controller also has logic in its handle_event() method!

Reply:
good catch, @nvidianz ^ Yan suggest we add this logic to XGBController as well.